libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/RequestHelper.java

```diff
@@ -125,27 +125,26 @@ public static String getJourneyParameter(URI journeyUri, String key) {
 
     public static String getScoreType(ProcessRequest request)
             throws HttpResponseExceptionWithErrorBody {
-        return extractValueFromLambdaInput(
-                request, "scoreType", ErrorResponse.MISSING_SCORE_TYPE, String.class);
+        return extractValueFromLambdaInput(request, "scoreType", ErrorResponse.MISSING_SCORE_TYPE);
     }
 
     public static Integer getScoreThreshold(ProcessRequest request)
             throws HttpResponseExceptionWithErrorBody {
         return extractValueFromLambdaInput(
-                request, "scoreThreshold", ErrorResponse.MISSING_SCORE_THRESHOLD, Integer.class);
+                request, "scoreThreshold", ErrorResponse.MISSING_SCORE_THRESHOLD);
     }
 
-    public static Boolean getIsUserInitiated(ProcessRequest request)
+    public static boolean getIsUserInitiated(ProcessRequest request)
             throws HttpResponseExceptionWithErrorBody {
-        return extractValueFromLambdaInput(
-                request,
-                "isUserInitiated",
-                ErrorResponse.MISSING_IS_USER_INITIATED_PARAMETER,
-                Boolean.class);
+        return Boolean.TRUE.equals(
```

Comment:
This doesn't hurt, but it looks like `extractValueFromLambdaInput` can never actually return `null` anyway?

Reply:
No it can't, it does a null check internally and throws. But Sonar doesn't detect that null check. Returning a primitive instead keeps it happy and means we don't have to do the null check everywhere we used the Boolean in the handler.

Reply:
Yeah ok - I guess the classic solution to this is to create specialisations for primitives, but this is fine. Those damn nulls...